src/Compilers/Core/Portable/Text/ChangedText.cs

```diff
@@ -357,8 +379,14 @@ private static ImmutableArray<TextChangeRange> Merge(ImmutableArray<TextChangeRa
                 // no more new changes, just add remaining old changes
                 while (oldIndex < oldChanges.Length)
                 {
-                    AddRange(list, oldChanges[oldIndex]);
+                    if (needNextOldChange)
+                    {
+                        oldChange = oldChanges[oldIndex];
+                    }
```

Comment:
📝 This was failing to handle the case where the first remaining old change was an adjusted change.